eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -81,16 +81,17 @@ object Graph {
         // for each neighbor
         currentNeighbors.foreach { edge =>
 
-          //  test for ignored edges
-          if (!(edge.update.htlcMaximumMsat.exists(_ < amountMsat) ||
-            amountMsat < edge.update.htlcMinimumMsat ||
-            ignoredEdges.contains(edge.desc))
-          ) {
+          val neighbor = edge.desc.b
 
-            val neighbor = edge.desc.b
+          // note: 'cost' contains the smallest known cumulative cost (amount + fees) necessary to reach 'current' so far
+          // note: there is always an entry for the current in the 'cost' map
+          val newMinimumKnownCost = edgeWeight(edge, cost.get(current.key), neighbor == targetNode)
 
-            // note: the default value here will never be used, as there is always an entry for the current in the 'cost' map
-            val newMinimumKnownCost = cost.get(current.key) + edgeWeightByAmount(edge, amountMsat)
+          // test for ignored edges
+          if (!(edge.update.htlcMaximumMsat.exists(_ < newMinimumKnownCost) ||
```

Comment:
is that the most optimal order for the testing conditions? Given that the first to return `true` will return immediately

Reply:
done